dev/tests/api-functional/framework/Magento/TestFramework/Assert/AssertArrayContains.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\TestFramework\Assert;
+
+use PHPUnit\Framework\Assert;
+
+/**
+ * Check that actual data contains all values from expected data
+ * But actual data can have more values than expected data
+ */
+class AssertArrayContains
```

Comment:
Does not look good to me, reminds https://github.com/sebastianbergmann/phpunit/pull/1648.

Necessity of such assert usually means test aims to verify too much.

Why it's not enough to use:
- `assertEquals($expectedValue, $array['key'])` (will `E_NOTICE` break test execution or just fail this particular test method?)
- `assertEquals($expectedArray, array_intersect_key($expectedArray, $actualArray))`

?

Reply:
In this assert, we provide possibility to work with nested arrays

Reply:
Is it a good idea to compare such arrays? Something really common for WebAPI?

Reply:
Basically, my main concern is that whether tests with such assert will be accurate enough if they ignore some junk in data.

Reply:
It could be useful in WebAPI tests
For example, we create entity with predefined data (only base fields) via WebAPI request
Next, we get created object data via GET request.
But obtained data could contain additional fields like 'extension_attributes'

Reply:
Ok. Special thanks for not making it available in unit tests 👍

Reply:
Yes
It is trick only for WebApi functional tests